subsys/fs/shell.c

```diff
@@ -334,6 +334,62 @@ static int cmd_read(const struct shell *shell, size_t argc, char **argv)
 	return 0;
 }
 
+static int cmd_cat(const struct shell *shell, size_t argc, char **argv)
+{
+	char path[MAX_PATH_LEN];
+	struct fs_dirent dirent;
+	struct fs_file_t file;
+	int err;
+
+	fs_file_t_init(&file);
+
+	for (size_t i = 1; i < argc; ++i) {
+		create_abs_path(argv[i], path, sizeof(path));
+
+		err = fs_stat(path, &dirent);
+		if (err < 0) {
+			shell_error(shell, "Failed to obtain file %s (err: %d)",
+					path, err);
+			continue;
+		}
+
+		if (dirent.type != FS_DIR_ENTRY_FILE) {
+			shell_error(shell, "Note a file %s", path);
+			continue;
+		}
+
+		err = fs_open(&file, path, FS_O_READ);
+		if (err) {
+			shell_error(shell, "Failed to open %s (%d)", path, err);
+			continue;
+		}
+
+		while (true) {
+			ssize_t read;
+			uint8_t buf[BUF_CNT];
+
+			read = fs_read(&file, buf, sizeof(buf));
+			if (read == 0) {
+				break;
+			}
+
+			if (read < 0) {
+				shell_error(shell, "Failed to read from file %s (err: %d)",
+					path, read);
+				break;
+			}
```

Comment:
You can break here every time `read <= 0`, and then report error when `read < 0` outside of the loop.
Currently you have two checks in each loop run that will be false most of the time.

Reply:
Thanks! fixed.